pkg/codegen/schema.go

```diff
@@ -208,11 +210,34 @@ func GenerateGoSchema(sref *openapi3.SchemaRef, path []string) (Schema, error) {
 
 	// Check for custom Go type extension
 	if extension, ok := schema.Extensions[extPropGoType]; ok {
-		typeName, err := extTypeName(extension)
+		typeDetails, err := extTypeName(extension)
 		if err != nil {
 			return outSchema, fmt.Errorf("invalid value for %q: %w", extPropGoType, err)
 		}
-		outSchema.GoType = typeName
+
+		outSchema.GoType = typeDetails.Type
+		// if we have a custom import for our type as it is an external type we need to the imports
+		if typeDetails.Import != "" {
+			// if the import should have an alias it needs to be specified
+			if typeDetails.Alias != "" {
+				// we need to set the gotype with the correct import name
+				outSchema.GoType = fmt.Sprintf("%s.%s", typeDetails.Alias, typeDetails.Type)
+				outSchema.CustomImports = append(outSchema.CustomImports, fmt.Sprintf("%s \"%s\"", typeDetails.Alias, typeDetails.Import))
+			} else {
+				// as no alias is provided we need to take the import
+
+				// if there is an import name specified instead of an alias, take it
+				if strings.Contains(typeDetails.Import, ":") {
+					splitImport := strings.Split(typeDetails.Import, ":")
```

Comment:
Might as well be using `strings.Cut` here.
Would enable you to avoid this `strings.Contains` too, since you could len-check on `strings.Cut`